docs/source/configuration.rst

```diff
@@ -294,6 +294,11 @@ QueryParameter for key ``choices_param``, whose 'remote' name is
 
 will request the resource at URL http://example.com?ChoicesParam=value
 
+For a BodyParameter, it is allowed that the name attribute has value None. In that
```

Comment:
The following is **not** something you have to address as part of this PR **so consider it FYI**

* This update to the documentation makes me realize something, if you can supply `name=None`, should that make `name` an optional parameter?
* So to provide a custom, non-dictionary body, the user has to provide `name=None`. That doesn't seem to be an obvious, or telling, way to instruct the use of a non-dictionary body. What about creating a new subclass of `BodyParameter` that doesn't have a `name` parameter (and internally uses `None` as the value of `name`), e.g. `AnybodyParameter` - I'm just kidding about the name, but the suggestion stands ;).

Reply:
I agree that this approach doesn't clearly capture what a `name=None` enables. I see it as a first step to give the qrest user the ability to wrap endpoints with more complex body structures (e.g. non-dictionary and nested dictionaries (which was already possible, but only the root keys can be added to a resource as BodyParameter)). 

In a second phase, I'd like to improve the BodyParameter to support json schemas. The body that the enduser provides can be validated with this schema, and the schema can be used to help the end user building the body, e.g. by making a template available for them to fill.